internal/modifier/gated.go

```diff
@@ -78,5 +79,22 @@ func NewFeatureGatedModifier(logger logger.Interface, cfg *config.Config, image
 		discoverers = append(discoverers, d)
 	}
 
+	if !cfg.Features.AllowCUDACompatLibsFromContainer.IsEnabled() {
+		compatLibHookDiscoverer := discover.NewCUDACompatHookDiscoverer(logger, cfg.NVIDIACTKConfig.Path, driver)
+		discoverers = append(discoverers, compatLibHookDiscoverer)
+		if cfg.NVIDIAContainerRuntimeConfig.Mode == "legacy" {
+			ldcacheIpdateHookDiscoverer, err := discover.NewLDCacheUpdateHook(
```

Comment:
Note: Since the `nvidia-container-runtime-hook` is invoked as a `prestart` hook, this is done BEFORE the `createContainer` hook that we insert above. This means that we need to once again run the update ldcache hook.

Reply:
If we create the file in `nvidia-container-runtime-hook`, then wouldn't that remove the need to add the `cuda-compat` createContainer hook altogether?

Reply:
I did try it, and my initial assumption was incorrect. Because we invoke ldconfig as:
```
        argv = (char * []){cnt->cfg.ldconfig, "-f", "/etc/ld.so.conf", "-C", "/etc/ld.so.cache", cnt->cfg.libs_dir, cnt->cfg.libs32_dir, NULL};
```
in `libnvidia-container`, this means that the `cnt->cfg.libs_dir` and `cnt->cfg.libs32_dir` folders *take precedence* over the files in `/etc/ld.so.conf.d` and the CUDA libraries present there are used.

We could rework the `libnvidia-container` implementation further, but the intent of this change is also to provide the functionality for #910 so that we can deprecate the legacy code path by default.

Reply:
Got it. So for legacy mode, we are adding two createContainer hooks ( the `cuda-compat` hook and the `update-ldcache`) to ensure the compat libs are used, the caveat being that we execute ldconfig twice.

Reply:
Yes, that is correct.